### PR TITLE
lib/fs: Cache user lookups

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

Possibly make Windows scanning faster again.

### CODE REVIEW

**Comment by imsodin** [comment]
Taking an hour for a user/group change to be synced - that seems like a pretty big annoyance if I were to use that feature, isn't it?

**Comment by calmh** [comment]
That's not what it caches though, it caches the account name that belongs to a UID/SID. So yeah if you rename a user account it will take a while to be reflected, but more so because we won't detect it as a change at all...

**Comment by imsodin** [comment]
> That's not what it caches though, it caches the account name that belongs to a UID/SID. So yeah if you rename a user account it will take a while to be reflected, but more so because we won't detect it as a change at all...

Ou dang, I really need to review the original change first - I clearly am not up to speed here.

File: lib/fs/platform_common.go [hunk 1]
```diff
@@ -53,3 +55,39 @@ func unixPlatformData(fs Filesystem, name string) (protocol.PlatformData, error)
 		},
 	}, nil
 }
+
+type cacheEntry[T any] struct {
+	value T
+	when  time.Time
+}
+
+type valueCache[T any] struct {
```
Comment (AudriusButkevicius) [inline lib/fs/platform_common.go:64]: make the key generic?

**Review by AudriusButkevicius — APPROVED** [review]

**Pull request merged**